core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -565,10 +583,11 @@ public void forceAvailabilityMode(String cacheName, AvailabilityMode availabilit
       public void handleViewChange(final ViewChangedEvent e) {
          // Need to recover existing caches asynchronously (in case we just became the coordinator).
          // Cannot use the async notification thread pool, by default it only has 1 thread.
-         asyncTransportExecutor.submit(new Runnable() {
+         viewHandlingCompletionService.submit(new Callable<Void>() {
```

Comment:
I have this rule in my head: I use ```Callable``` when a return value is expected, otherwise I use ```Runnable```. And since the original code was a ```Runnable``` I would like to keep it :)